livekit-agents/livekit/agents/job.py

```diff
@@ -143,9 +145,24 @@ def agent(self) -> rtc.LocalParticipant:
         return self._room.local_participant
 
     def add_shutdown_callback(
-        self, callback: Callable[[], Coroutine[None, None, None]]
+        self,
+        callback: Union[
+            Callable[[], Coroutine[None, None, None]],
+            Callable[[str], Coroutine[None, None, None]],
+        ],
     ) -> None:
-        self._shutdown_callbacks.append(callback)
+        """
+        Add a callback to be called when the job is shutting down.
+        Optionally the callback can take a single argument, the shutdown reason.
+        """
+        if callback.__code__.co_argcount > 0:
```

Comment:
FYI I think this misses the case where you pass in a reference to a bound function w/ `self` first arg